### PR TITLE
Rewrote grammar rules to support loops and interpolation

### PR DESCRIPTION
**Pull request by Autalyst**
Closes Issue #7

### CODE REVIEW

File: decklang/decklang.ne [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
-@include "plugins/text.ne"
-@include "plugins/font.ne"
-@include "primitives_advanced.ne"
-@include "primitives.ne"
+Program -> Expressions | null
+Expressions -> Expression Expressions
+Expression -> Loop | VariableDeclaration | Directive | Comment | NewLine
 
-main -> _ (line_of_code _):*
+Comment -> "`" btstring
```
Comment (seiyria) [inline decklang/decklang.ne]: no need to have comments, they get ripped out before the program sees them

File: decklang/decklang.ne [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+LoopStart -> "loop" _ "=" _ "<" _ Assignment _ "> to " LoopGoal
+LoopGoal -> Integer | Variable | LoopVariable
+LoopBody -> LoopEnd | Expression LoopBody
+LoopEnd -> "endloop"
+
+ArgumentList -> Argument | Argument _ "," _ ArgumentList
+Argument -> Identifier | String | Integer | HexString | Variable | LoopVariable
+Assignment -> Identifier _ "=" _ Argument
+
+Variable -> "[" _ Identifier _ "]"
```
Comment (seiyria) [inline decklang/decklang.ne]: no need to have this in the grammar, they're pulled out and replaced before seen by the program

File: decklang/primitives.ne [hunk 3]
```diff
@@ -1,43 +0,0 @@
-@builtin "number.ne"
-@builtin "string.ne"
-
-# commonly used helper functions
-@{% function nuller() { return null; } %}
-@{% function emptystring() { return ''; } %}
-@{% function joiner(d) { return d.join(''); } %}
```
Comment (seiyria) [inline decklang/primitives.ne]: these 3 functions are useful for id functions

**Comment by Autalyst** [comment]
I'm going to rework this with my new understanding of how this works

**Pull request closed without merging**